biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotationReader.java

```diff
@@ -139,9 +134,9 @@
 
 		@Override
 		public void annotation(Annotation annotation) throws Exception {
-			java.lang.annotation.Annotation a = annotation.getAnnotation();
-
-			if (a instanceof Beans) {
+			String fqn = Descriptors.binaryToFQN(annotation.getName()
+				.getBinary());
+			if (fqn.equals("org.osgi.service.cdi.annotations.Beans")) {
```

Comment:
By no means critical, but would it be cleaner to have these as `static final String`s?

Reply:
That type name should never change. :-) So parking it in a constant is not so important.

Reply:
they could...

Reply:
since there's a regression bug here let's get the fix in and then I'll optimize.